Apply land use changes correctly (and with extra checking)

MartinDix opened the pull request:
Fixes problems described in #22.

blimlim commented on scripts/update_landuse.py:
```diff
@@ -42,14 +43,19 @@ def transform(self, source, result):
+month = mf.fixed_length_header.t2_month
+day = mf.fixed_length_header.t2_day
+if month != 1 or day != 1:
+    raise Exception(f"Unexpected month, day in update_landuse.py {month} {day}")
```
Just a small idea but I was wondering if would be good to have a more specific exception like a `ValueError` here.

MartinDix replied:
Good idea

blimlim reviewed (approved):
Changes look good to me – looks like a lot of detective work. I just added some questions for clarifying a couple of the changes.

One other question that came up while reviewing: `update_landuse.py` currently has
```python
landuse = xarray.open_dataset('work/atmosphere/INPUT/cableCMIP6_LC_1850-2015.nc').fraction
```
Will this script be reused for other configurations? I was wondering whether we should remove the hard coded filepath if so.

MartinDix commented on config.yaml:
```diff
@@ -60,8 +63,8 @@ calendar:
 runspersub: 1
 
 userscripts:
-    # Setup land use changes
-    setup: ./scripts/pre.sh work/atmosphere/INPUT/cableCMIP6_LC_1850-2015.nc
+    # Apply land use changes after each run
+    run: ./scripts/update_landuse_driver.sh
```
Yes

MartinDix commented:
> Changes look good to me – looks like a lot of detective work. I just added some questions for clarifying a couple of the changes.
> 
> One other question that came up while reviewing: `update_landuse.py` currently has
> 
> ```python
> landuse = xarray.open_dataset('work/atmosphere/INPUT/cableCMIP6_LC_1850-2015.nc').fraction
> ```
> 
> Will this script be reused for other configurations? I was wondering whether we should remove the hard coded filepath if so.

Different land use files would be used for future scenarios. Moved the filename to the driver script so it's passed to the python script as an argument.
